### DISCUSSION TITLE
Step by step installation with docker / portainer - guide / questions?

### DISCUSSION THREAD

**Discussion by gauiPPP** [getredash/redash#6444] [Q&A] [answered]
My first installation of redash was with the provided script which ran the functions:
install_docker
create_directories
create_config
setup_compose
https://github.com/getredash/setup/blob/master/setup.sh

Now I have studied a bit more docker and would like to run different services and programs, and manage them through Portainer.
I have pulled the latest redash docker image and it is running with port 5000 open. When accessing the port I get an internal server error.
On portainer I have only portainer and redash containers running

Looking at my first installation on a dedicated VM (Ubuntu 18-04 LTS) I can see with "docker compose ps" these services:
NAME                        IMAGE                         COMMAND                  SERVICE             CREATED                                                                                                   STATUS              PORTS
redash_adhoc_worker_1       redash/redash:10.1.0.b50633   "/app/bin/docker-ent…"   adhoc_worker        2 weeks ago                                                                                               Up 2 weeks          5000/tcp
redash_nginx_1              redash/nginx:latest           "nginx -g 'daemon of…"   nginx               2 weeks ago                                                                                               Up 2 weeks          0.0.0.0:80->80/tcp, :::80->80/tcp, 443/tcp
redash_postgres_1           postgres:9.6-alpine           "docker-entrypoint.s…"   postgres            2 weeks ago                                                                                               Up 2 weeks          5432/tcp
redash_redis_1              redis:5.0-alpine              "docker-entrypoint.s…"   redis               2 weeks ago                                                                                               Up 2 weeks          6379/tcp
redash_scheduled_worker_1   redash/redash:10.1.0.b50633   "/app/bin/docker-ent…"   scheduled_worker    2 weeks ago                                                                                               Up 2 weeks          5000/tcp
redash_scheduler_1          redash/redash:10.1.0.b50633   "/app/bin/docker-ent…"   scheduler           2 weeks ago                                                                                               Up 2 weeks          5000/tcp
redash_server_1             redash/redash:10.1.0.b50633   "/app/bin/docker-ent…"   server              2 weeks ago                                                                                               Up 2 weeks          0.0.0.0:5000->5000/tcp, :::5000->5000/tcp
redash_worker_1             redash/redash:10.1.0.b50633   "/app/bin/docker-ent…"   worker              2 weeks ago                                                                                               Up 2 weeks          5000/tcp

Should I be running similar services also in portainer, each as their own container?

On the other hand shouldn't the redash docker container also install/start needed services which are defined under depends_on in the file docker-compose.yml?

the redash container accessed with console shows in docker-compose.yml:
services:
  server:
    build: .
    command: dev_server
    depends_on:
      - postgres
      - redis
    ports:
      - "5000:5000"
      - "5678:5678"

**Comment by guidopetri** [comment, marked as answer]
> Should I be running similar services also in portainer, each as their own container?

Yes - the redash image has several components which work together, namely, the server + workers. The redis + postgres + nginx containers can be exchanged for another redis/database/reverse proxy container. You'll need at least 1 server container and 1 (adhoc) worker, as well as 1 database + 1 redis container somewhere that Redash can access.

**Comment by celinefrancoise** [comment]
hello

**Reply by justinclift** [reply]
Hi, are you needing assistance with something? :smile:
